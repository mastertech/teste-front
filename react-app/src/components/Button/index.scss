@use "../../themes/variables.scss" as theme;

.button {
  display: flex;
  align-items: center;
  justify-content: center;
  background: none;
  border: none;
  padding: 10px;
  background-color: theme.$color-yellow-primary;
  box-shadow: theme.$shadow;
  color: black;
  @extend .text-lg;
  font-weight: theme.$font-bold;
  width: 100%;
  max-width: 100px;
  margin: theme.$margin-default;
  cursor: pointer;
  transition: transform 0.25s ease, box-shadow 0.25s ease,
    background-color 0.25s ease;
}

.button:hover {
  transform: translate3d(0px, -1px, 0px);
  box-shadow: 0 7px 14px rgba(50, 50, 93, 0.1), 0 3px 6px rgba(0, 0, 0, 0.08);
}

.donut-spinner {
  border: 4px solid white;
  border-left-color: black;
  border-radius: 50%;
  width: 20px;
  height: 20px;
  animation: donut-spin 1.2s linear infinite;
}

@keyframes donut-spin {
  to {
    transform: rotate(1turn);
  }
}
