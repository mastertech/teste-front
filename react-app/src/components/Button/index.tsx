import { ButtonHTMLAttributes, FC } from 'react'
import './index.scss'

type ButtonProps = ButtonHTMLAttributes<HTMLButtonElement> & {
  title: string
  isLoading: boolean
}

export const Button: FC<ButtonProps> = ({
  children,
  title,
  isLoading,
  ...props
}) => {
  return (
    <button className="button" {...props}>
      {isLoading ? <div className="donut-spinner" /> : title}
    </button>
  )
}
