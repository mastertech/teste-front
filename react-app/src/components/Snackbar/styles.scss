@use "../../themes/variables.scss" as theme;

.snackbar {
  visibility: visible;
  min-width: 250px;
  margin-left: -125px;
  background-color: theme.$color-darkest-gray;
  color: #fff;
  text-align: center;
  border-radius: 2px;
  padding: 16px;
  position: fixed;
  z-index: 1;
  left: 50%;
  bottom: 30px;
  animation: fadein 1.25s infinite alternate;
}

@keyframes fadein {
  from {
    bottom: 0;
    opacity: 0;
  }
  to {
    bottom: 50px;
    opacity: 1;
  }
}
