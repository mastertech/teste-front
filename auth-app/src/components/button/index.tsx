import React, { HTMLAttributes } from 'react';

import styles from './button.module.scss';

type Props = {
  isLoading?: boolean;
} & HTMLAttributes<HTMLButtonElement>;

const Button: React.FC<Props> = ({ children, isLoading = false, ...rest }) => {
  return (
    <button
      type="submit"
      className={`${styles.btnContainer} ${isLoading && styles.spinner}`}
      disabled={isLoading}
      {...rest}
    >
      {!isLoading && <span>{children}</span>}
    </button>
  );
};

export default Button;
