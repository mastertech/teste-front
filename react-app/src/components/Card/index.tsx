import { FC } from 'react'
import './styles.scss'

export const Card: FC = ({ children }) => {
  return <div className="container">{children}</div>
}
