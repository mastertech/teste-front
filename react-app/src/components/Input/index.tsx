import { FC, forwardRef, HTMLProps, InputHTMLAttributes } from 'react'
import './styles.scss'

type InputProps = HTMLProps<HTMLInputElement> & {
  title: string
}

export const Input = forwardRef<HTMLInputElement, InputProps>(
  ({ title, ...props }, ref) => {
    return (
      <div className="wrapper">
        <input
          className="floating-input"
          ref={ref}
          placeholder={title}
          {...props}
        />
        <span className="floating-label">{title}</span>
      </div>
    )
  }
)
