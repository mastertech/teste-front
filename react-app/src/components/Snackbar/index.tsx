import { FC } from 'react'
import './styles.scss'

type SnackBarProps = {
  title: string
}

export const SnackBar: FC<SnackBarProps> = ({ title }) => {
  return <div className="snackbar">{title}</div>
}
