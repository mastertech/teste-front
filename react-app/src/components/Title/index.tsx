import { FC } from 'react'
import './styles.scss'

export const Title: FC = ({ children }) => {
  return <h1 className="title">{children}</h1>
}
