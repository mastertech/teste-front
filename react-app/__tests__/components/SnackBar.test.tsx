import * as renderer from 'react-test-renderer'
import { SnackBar } from '../../src/components/Snackbar'

describe('<SnackBar /> component', () => {
  test('should renders correctly', async () => {
    const props = {
      title: 'Test',
    }
    const tree = renderer.create(<SnackBar {...props} />).toJSON()
    expect(tree).toMatchSnapshot()
  })
})
