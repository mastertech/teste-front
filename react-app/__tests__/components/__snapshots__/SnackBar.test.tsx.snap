// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<SnackBar /> component should renders correctly 1`] = `
<div
  className="snackbar"
>
  Test
</div>
`;
